p2p/src/sync/peer.rs

```diff
@@ -203,12 +209,36 @@ where
         Ok(())
     }
 
+    /// Delays the processing of a new block until it can be accepted by the chainstate (but not more than `max_clock_diff`).
+    /// This is needed to allow the local or remote node to have slightly inaccurate clocks.
+    /// Without it, even a 1 second difference can break block synchronization
+    /// because one side may see the new block as invalid.
+    async fn sync_clock(&self, block_timestamp: BlockTimestamp) {
```

Comment:
Should this be called something like `delay_new_block()` instead, because `sync_clock()` sounds like a time somewhere is getting set

Reply:
The `sync_clock` function was removed.